File: servers/dotnet/Application/Core/Blogs/Assemblers/BlogAssembler.cs
using System;
using Application.Blogs.DTOs;
using Application.Users.Assemblers;
using Domain;

namespace Application.Blogs.Assemblers;

public class BlogAssembler
{
    public BlogDto Assemble(Blog entity) 
    {
        return new BlogDto 
        {
            Id = entity.Id,
            UserId = entity.UserId,
            Title = entity.Title,
            Body = entity.Body
        };
    }

    public List<BlogDto> Assemble(List<Blog> entities)
    {
        return entities
            .Select(entity => Assemble(entity))
            .ToList();
    }

    public Blog Disassemble(BlogDto blogDto) 
    {
        return Blog.NewInstance(blogDto.Title, blogDto.Body, blogDto.UserId);
    }

    public Blog DisassembleInto(BlogDto blogDto, Blog entity)
    {
        entity.Title = blogDto.Title;
        entity.Body = blogDto.Body;
        return entity;
    }
}

File: servers/springboot/server/src/main/java/com/gabrielgavrilov/ohmyblog/assembler/BlogAssembler.java
package com.gabrielgavrilov.ohmyblog.assembler;

import com.gabrielgavrilov.ohmyblog.api.BlogDto;
import com.gabrielgavrilov.ohmyblog.core.blog.entity.Blog;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
@RequiredArgsConstructor
public class BlogAssembler {

    private final UserAssembler userAssembler;

    public BlogDto assemble(Blog entity) {
        return new BlogDto()
                .setBlogId(entity.getBlogId())
                .setTitle(entity.getTitle())
                .setBody(entity.getBody())
                .setCreatedAt(entity.getCreatedAt())
                .setUserDto(userAssembler.assemble(entity.getUser()));
    }

    public Blog disassemble(BlogDto blogDto, UUID userId) {
        return Blog.newInstance(
                userId,
                blogDto.getTitle(),
                blogDto.getBody()
        );
    }

    public Blog disassembleInto(BlogDto blogDto, Blog entity) {
        return entity
                .setTitle(blogDto.getTitle())
                .setBody(blogDto.getBody());
    }

}
